include/exadg/incompressible_navier_stokes/postprocessor/line_plot_calculation_statistics_homogeneous.cpp

```diff
@@ -314,44 +316,20 @@ LinePlotCalculatorStatisticsHomogeneous<dim, Number>::setup(
       }
     }
 
-    create_directories(data.line_data.directory, mpi_comm);
+    create_directories(data.directory, mpi_comm);
   }
 }
 
 template<int dim, typename Number>
 void
-LinePlotCalculatorStatisticsHomogeneous<dim, Number>::evaluate(
-  VectorType const &   velocity,
-  VectorType const &   pressure,
-  double const &       time,
-  unsigned int const & time_step_number)
+LinePlotCalculatorStatisticsHomogeneous<dim, Number>::evaluate(VectorType const & velocity,
+                                                               VectorType const & pressure,
+                                                               bool const         write_statistics)
 {
-  if(data.statistics_data.calculate)
-  {
-    // EPSILON: small number which is much smaller than the time step size
-    double const EPSILON = 1.0e-10;
-
-    if((time > data.statistics_data.sample_start_time - EPSILON) &&
-       (time < data.statistics_data.sample_end_time + EPSILON) &&
-       (time_step_number % data.statistics_data.sample_every_timesteps == 0))
-    {
-      // evaluate statistics
-      do_evaluate(velocity, pressure);
+  do_evaluate(velocity, pressure);
 
-      // write intermediate output
-      if(time_step_number % data.statistics_data.write_output_every_timesteps == 0)
-      {
-        do_write_output();
-      }
-    }
-
-    // write final output
-    if((time > data.statistics_data.sample_end_time - EPSILON) && write_final_output)
-    {
-      do_write_output();
-      write_final_output = false;
-    }
-  }
+  if(write_statistics)
```

Comment:
Suggestion 1: Write 

```c++
if(time_control.get_counter() % write_preliminary_results_every_nth_evaluation == 0 or 
   time_control.reached_end_time())
  do_write_output();
```

`TimeControlStatistics` should not implement a function called `needs_evaluation()` in my opinion. With this suggestion, you might need to adjust the if-statement in the postprocessor
```c++
if(time_control.needs_evaluation() or time_control.reached_end_time())
  postprocessing_module->evaluate();
```

This implements the logic described here https://github.com/exadg/exadg/pull/262#discussion_r1001521153.

Suggestion 2: The alternative is to completely separate the functionality of `do_evaluate()` and `write_preliminary_results(). This would lead to

```c++
if(time_control.needs_evaluation())
  postprocessing_module->evaluate();

if(time_control_statistic.write_results())
  postprocessing_module->write_results();
```